java/ql/src/semmle/code/java/security/OgnlInjection.qll

```diff
@@ -0,0 +1,124 @@
+/** Provides classes to reason about OGNL injection vulnerabilities. */
+
+import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.ExternalFlow
+
+/**
+ * A data flow sink for unvalidated user input that is used in OGNL EL evaluation.
+ *
+ * Extend this class to add your own OGNL injection sinks.
+ */
+abstract class OgnlInjectionSink extends DataFlow::Node { }
+
+/**
+ * A unit class for adding additional taint steps.
+ *
+ * Extend this class to add additional taint steps that should apply to the `OgnlInjectionFlowConfig`.
+ */
+class OgnlInjectionAdditionalTaintStep extends Unit {
+  /**
+   * Holds if the step from `node1` to `node2` should be considered a taint
+   * step for OGNL injection taint configurations.
+   */
+  abstract predicate step(DataFlow::Node node1, DataFlow::Node node2);
+}
+
+private class DefaultOgnlInjectionSinkModel extends SinkModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.apache.commons.ognl;Ognl;false;getValue;;;Argument[-1..0];ognl-injection",
```

Comment:
All of these `Argument[-1..0]` seem a bit lazy - is it really the case that both argument 0 and the qualifier are sinks for all of these methods?

Reply:
You are right, something got lost in translation when converting from regular CodeQL classes to CSV models. Fixed in 
2628d3d, thanks!